.github/workflows/python-package.yml

```diff
@@ -0,0 +1,44 @@
+# This workflow will install Python dependencies, run tests and lint with a variety of Python versions
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-python
+
+name: Python package
+
+on:
+  # run for changes in main branch
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+  # also run weekly: Fridays at 11:47.
+  schedule:
+    - cron: "47 11 * * 6"
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ["3.7", "3.8", "3.9", "3.10"]
```

Comment:
As Python 3,7 reaches EndoOfLife in 2 weeks and 3.11 is already out, I would propose to change the versions accordingly. To make sure we test with supported versions.